libs/cypress-indexeddb/src/lib/alias-setup.ts

```diff
@@ -0,0 +1,48 @@
+const STORES = new Map<string, IDBObjectStore>();
+const DATABASES = new Map<string, IDBDatabase>();
+
+type IDBItemType = 'store' | 'database';
+
+export function overrideAs(
+  originalAs: (subject: any, alias: string) => unknown,
+  subject: any | IDBObjectStore,
+  alias: string
+) {
+  if (subject?.constructor?.name === 'IDBObjectStore') {
+    STORES.set(alias, subject);
+    return subject;
+  } else if (subject?.constructor?.name === 'IDBDatabase') {
+    DATABASES.set(alias, subject);
+    return subject;
+  } else {
+    return originalAs(subject, alias);
+  }
+}
```

Comment:
we could get rid of those `any`s easily
```suggestion
const isIDBObjectStore = (subject: unknown): subject is IDBObjectStore => {
  return typeof subject === 'object' && subject?.constructor?.name === 'IDBObjectStore'
}

const isIDBDatabase = (subject: unknown): subject is IDBDatabase => {
  return typeof subject === 'object' && subject?.constructor?.name === 'IDBDatabase'
}

export function overrideAs(
  originalAs: (subject: unknown, alias: string) => unknown,
  subject: unknown,
  alias: string
) {
  if (isIDBObjectStore(subject)) {
    STORES.set(alias, subject);
    return subject;
  } else if (isIDBDatabase(subject)) {
    DATABASES.set(alias, subject);
    return subject;
  } else {
    return originalAs(subject, alias);
  }
}
```